packages/dds/merge-tree/src/partialLengths.ts

```diff
@@ -314,25 +371,34 @@ export class PartialSequenceLengths {
         combinedPartialLengths: PartialSequenceLengths,
         segment: ISegment,
         removalInfo?: IRemovalInfo) {
+        const isLocal = (removalInfo === undefined && segment.seq === UnassignedSequenceNumber)
+            || (removalInfo !== undefined && segment.removedSeq === UnassignedSequenceNumber);
         // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        let seq = segment.seq!;
+        let seq = isLocal ? segment.localSeq! : segment.seq!;
```

Comment:
still going through this, but this seems odd, would at least like a comment about why this is safe to do, as seq and local seq cannot be evaluated against each other in any context, as they are independent

Reply:
if you want to make them comparable you need to do something like collabwindow.currentSeq + local seq, that computes a synthetic future seq that is good for the lifetime of the current currentSeq

Reply:
I added a note to the doc comment of the function that should be helpful in clarifying what's going on here. Also did the variable renames you suggested which make this less alarming (seq -> seqOrLocalSeq).